lib/Parse/Lexer.cpp

```diff
@@ -2530,7 +2525,10 @@ Token Lexer::getTokenAtLocation(const SourceManager &SM, SourceLoc Loc,
   return L.peekNextToken();
 }
 
-void Lexer::lexTrivia(ParsedTrivia &Pieces, bool IsForTrailingTrivia) {
+StringRef Lexer::lexTrivia(bool IsForTrailingTrivia) {
```

Comment:
We don't need to return `StringRef` here? `TrailingTrivia` can also like:
```
const char *trailingTriviaStart = CurPtr
lextTrivia(true);
TrailingTrivia = StringRef(trailingTriviaStart, CurPtr - trailingTriviaStart);
```

Reply:
We could, but we'd only be creating the `StringRef` at the call site to store it in `LeadingTrivia`. Without measuring, I also assume that constructing and returning a `StringRef` also isn't particularly expensive, so I’d rather have a cleaner API here. Or do you have a different feeling about this?

Reply:
Merging now to get this in. If you have a feeling that this will improve performance or anything else, I can investigate in a follow-up PR.

Reply:
I'd just like to keep the consistency between `LeadingTrivia` and `TrailingTrivia`.
Another design would be `lexTrivia` receive the pointer and form the trivia inside it:
```cpp
void lexTrivia(bool IsForTrailingTrivia, const char *TriviaStart, StringRef &Trivia) {
  // Advance 'CurPtr' to the end of the trivia ...

  Trivia = StringRef(startPtr, CurPtr - startPtr)
}
```
I'll put up a PR when I'm available.